frontend/src/metabase-lib/metadata/utils/fields.ts

```diff
@@ -45,6 +45,12 @@ export function getUniqueFieldId({
   const fieldIdentifier = getFieldIdentifier({ id, name });
 
   if (isVirtualCardId(table_id)) {
+    const isVirtual = typeof id !== "number";
+
+    if (isVirtual && name) {
```

Comment:
I had to add `isVirtual` part to this condition, otherwise 4 e2e tests were failing.